### ISSUE TITLE
The program just stops. It seems that the problem is with this line "_lock.acquire(timeout)"

### ISSUE THREAD

**Issue by Sunnydreamrain** [mila-udem/platoon#67]
Hi all,

After running a few hours, the program just stops. It seems that the problem is with this line "_lock.acquire(timeout)". As shown in "https://github.com/mila-udem/platoon/blob/master/platoon/channel.py#L390", the timeout is "None" as default.

From here (http://semanchuk.com/philip/posix_ipc/), "A timeout of None (the default) implies no time limit. The call will not return until its wait condition is satisfied."
I don't know what the problem is yet.

Anyone has encountered this before?

**Comment by mgermain** [comment]
Everything you said is right but what do yu mean it just stops?
Did you start both the worker and the controller?
Did you redirect the output? Are they running in the background on in different shells?
Are you using the launcher instead?

**Comment by Sunnydreamrain** [comment]
I am using the launcher. I did not redirect the output. 
It runs correctly for a while, sometimes even over 5000 batches (sometimes around 2000 batches.). Then it just stops (the process is gone). There is no error in controller or worker err and out files.

In order to locate where the error happens, I added a few print lines in the code. 
def sync_params(self, synchronous=True):
        print('actually start sync')
        if synchronous:
            print('start lock')
            self.lock_params()
            print('lock done')
        print('start update')
        self.update_fn(self.shared_params)
        print('update done')
        if synchronous:
            print('start unlock')
            self.unlock_params()
            print('unlock done')

    def lock_params(self, timeout=None):
        print ('start lock acquire')
        test=self._lock.acquire(timeout)
        print(test)
        print ('lock acquired')

I found that when the error happens, the fours worker processes are at the following location.
Two of then just going through "self.unlock_params()". printed "unlock done"
One is start to acquire the lock. printed "start lock acquire"
One is updating. printed "start update"

I am using a machine with 2 CPUs (Intel®Xeon E5-2620 V4) and 4 GPUS (TitanX). The system is Ubuntu server. I tried both python 2.7 and python 3. I also tried using only two gpus, but the error still occurs.

Any advice?

Thanks a lot.
